case-lib/user.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+# function func_check_version uses dpkg to get the package version
+# and compares it with the specified version. If the package version
+# is less than the specified version, it means the version dependency
+# is not met.
+# Args: $1: package name
+#       $2: specifiedversion
+func_check_version()
```

Comment:
> To prepare the code for other distributions you could use `sort -V` https://unix.stackexchange.com/questions/285924/how-to-compare-a-programs-version-in-a-shell-script

Do you mean something like:
```
    if [ "$(printf '%s\n' "$2" "$ver" | sort -V | head -n1)" = "$ver" ]; then
        die "$1 version $ver doesn't meet the requirement >= $2"
    fi
```

Reply:
Yes I meant something looking like this.

I recommend a new "version_greater_or_equal()" or "version_ge()" function that would not `die` but just `test` and that would be OS-independent. This would make it easier to port to ChromeOS or other later.

Reply:
> Yes I meant something looking like this.
> 
> I recommend a new "version_greater_or_equal()" or "version_ge()" function that would not `die` but just `test` and that would be OS-independent. This would make it easier to port to ChromeOS or other later.

@marc-hb But we still need `dpkg -l` to get the package version. So it seems to be hard be OS-independent. `aplay --version` can get the aplay version. But the version is not accurate enough. And it seems hard to get the alsa-lib version without `dpkg`

Reply:
I wrote "easier to make OS-independent LATER", not now not in this PR. 

Never duplicate any code, that's why we have functions. The suggestion is: `main() -> check_version() -> compare_version()`

The 3rd one is OS-independent and does not die. The 2nd one is `dpkg` specific (for now) and it may die.

Reply:
The code in https://github.com/thesofproject/sof-test/pull/600/commits/c626edba79f1985c is basically the opposite of what I just recommended. Can you please not "resolve conversation"?